packages/node/src/indexer/project.service.ts

```diff
@@ -105,6 +122,44 @@ export class ProjectService {
         await this.poiService.init(this.schema);
       }
     }
+
+    // bestBlocks
+    const startBestBlocks = await this.getMetadataBestBlocks();
+    const lastFinalizedVerifiedHeight =
+      await this.getLastFinalizedVerifiedHeight();
+    if (argv['best-block']) {
+      this._startBestBlocks = startBestBlocks ?? {};
+      this.bestBlockService.init(
+        this.metadataRepo,
+        this.startBestBlocks,
+        lastFinalizedVerifiedHeight,
+      );
+    } else {
+      // Has previous indexed with bestBlocks, but discontinue to use best block in this run
+      if (
+        (startBestBlocks !== undefined &&
+          Object.entries(startBestBlocks).length !== 0) ||
+        lastFinalizedVerifiedHeight !== null
+      ) {
+        if (lastFinalizedVerifiedHeight < this._startHeight) {
+          logger.info(
+            `Found un-finalized block from previous indexing but unverified, discontinued fetch from un-finalized in this run will require to rollback to last finalized block ${lastFinalizedVerifiedHeight} `,
+          );
+          await this.reindex(lastFinalizedVerifiedHeight);
+          this._startHeight = lastFinalizedVerifiedHeight;
+          logger.info(
+            `Successful rewind to block ${lastFinalizedVerifiedHeight} !`,
+          );
+        } else {
+          // if finalized block haven't been process yet, then remove best blocks from both metadata and memory
+          const transaction = await this.sequelize.transaction();
+          await this.storeService.resetBestBlocks(transaction);
+          await this.storeService.resetLastFinalizedVerifiedHeight(transaction);
+          await transaction.commit();
+          this.bestBlockService.resetBestBlocks();
+        }
+      }
+    }
```

Comment:
It would be nice if we could move as much as possible into the best block service.